nomad/state/state_store.go

```diff
@@ -1080,6 +1079,70 @@ func deleteNodeFromPlugin(txn *memdb.Txn, plug *structs.CSIPlugin, node *structs
 	return nil
 }
 
+// deleteJobFromPlugin removes allocations from this job from the plugins those jobs are
```

Comment:
This might read a little better as:

```go
// deleteJobFromPlugin removes the allocations of this job from any plugins the job is
// running, possibly deleting the plugin if its no longer in use. It's called in DeleteJobTxn
func (s *StateStore) deleteJobFromPlugin(index uint64, txn *memdb.Txn, job *structs.Job) error {
```

Reply:
👍